arkouda/dataframe.py

```diff
@@ -1509,6 +1509,103 @@ def groupby(self, keys, use_series=True):
 
         return self.GroupBy(keys, use_series)
 
+    @typechecked
+    def isin(self, values: Union[pdarray, Dict, Series, DataFrame]) -> DataFrame:
+        """
+        Determine whether each element in the DataFrame is contained in values.
+
+        Parameters
+        __________
+        values : pdarray, dict, Series, or DataFrame
+            The values to check for in DataFrame. Series can only have a single index.
+
+        Returns
+        _______
+        DataFrame
+            Arkouda DataFrame of booleans showing whether each element in the DataFrame is contained in values
+
+        See Also
+        ________
+        ak.Series.isin
+
+        Notes
+        _____
+        - Pandas supports values being an iterable type. In arkouda, we replace this with pdarray
+        - Pandas supports ~ operations. Currently, ak.DataFrame does not support this.
+
+        Examples
+        ________
+        >>> df = ak.DataFrame({'col_A': ak.array([7, 3]), 'col_B':ak.array([1, 9])})
+        >>> df
+            col_A  col_B
+        0      7      1
+        1      3      9 (2 rows x 2 columns)
+
+        When `values` is a pdarray, check every value in the DataFrame to determine if it exists in values
+        >>> df.isin(ak.array([0, 1]))
+           col_A  col_B
+        0  False   True
+        1  False  False (2 rows x 2 columns)
+
+        When `values` is a dict, the values in the dict are passed to check the column indicated by the key
+        >>> df.isin({'col_A': ak.array([0, 3])})
+            col_A  col_B
+        0  False  False
+        1   True  False (2 rows x 2 columns)
+
+        When `values` is a Series, each column is checked if values is present positionally. This means that for `True`
+        to be returned, the indexes must be the same.
+        >>> i = ak.Index(ak.arange(2))
+        >>> s = ak.Series(data=[3, 9], index=i)
+        >>> df.isin(s)
+            col_A  col_B
+        0  False  False
+        1  False   True (2 rows x 2 columns)
+
+        When `values` is a DataFrame, the index and column must match. Note that 9 is not found because the column
+        name does not match.
+        >>> other_df = ak.DataFrame({'col_A':ak.array([7, 3]), 'col_C':ak.array([0, 9])})
+        >>> df.isin(other_df)
+            col_A  col_B
+        0   True  False
+        1   True  False (2 rows x 2 columns)
+        """
+        if isinstance(values, pdarray):
+            # flatten the DataFrame so single in1d can be used.
+            flat = concatenate(list(self.data.values()))
+            lens = array([self.data[col].size for col in self.columns])
+            segs = concatenate([cumsum(lens) - lens, array([flat.size])])
+            df_def = {col: in1d(flat, values)[segs[i]:segs[i + 1]] for i, col in enumerate(self.columns)}
```

Comment:
I'm still a bit nervous about `in1d` being recomputed in the dict comprehension, to avoid risking it, maybe we do 
```python
flat_in1d = in1d(concatenate(list(self.data.values())), values)
segs = concatenate([array([0]), cumsum(array([self.data[col].size for col in self.columns]))])
df_def = {col: flat_in1d[segs[i]:segs[i + 1]] for i, col in enumerate(self.columns)}
```

Reply:
@pierce314159 I flipped this back.